contractcourt/channel_arbitrator.go

```diff
@@ -424,18 +424,24 @@ func (c *ChannelArbitrator) relaunchResolvers() error {
 	commitHash := contractResolutions.CommitHash
 
 	// Reconstruct the htlc outpoints and data from the chain action log.
-	// The purpose of the constructed htlc map is to supplement to resolvers
-	// restored from database with extra data. Ideally this data is stored
-	// as part of the resolver in the log. This is a workaround to prevent a
+	// The purpose of the constructed htlc map is to supplement to
+	// resolvers restored from database with extra data. Ideally this data
+	// is stored as part of the resolver in the log. This is a workaround
+	// to prevent a
 	// db migration.
 	htlcMap := make(map[wire.OutPoint]*channeldb.HTLC)
-	chainActions, err := c.log.FetchChainActions()
-	if err != nil {
-		log.Errorf("unable to fetch chain actions: %v", err)
-		return err
-	}
-	for _, htlcs := range chainActions {
-		for _, htlc := range htlcs {
+	for _, htlcs := range c.activeHTLCs {
```

Comment:
Make this change in the same commit where we stop writing chain actions?

Reply:
I think it stands on its own as: if we've already written chain actions for this channel, then the return parameter will be properly populated.

Reply:
But doesn't that mean that the commit where you stopped writing chain actions broke this part of the code?

Reply:
The diff is to be taken as a whole (all the commits), if a user cherry picks some commits, but not others, then behavior will be broken. As I said earlier, the commits are constructed so things compile, but if you run things like the itests at any commit but the end, they may fail.

Reply:
I do it differently and only create commits that have no broken behaviour. I also try to get the unit tests and itests running. If that seems to be work that is reverted again in later commits, it is an indication that my commit structure isn't right. But it is something up to preference too.